flight/vision/camera/camera.py

```diff
@@ -1 +1,179 @@
+import cv2
+import os
+import yaml
+import time
+from datetime import datetime
+import logging
+from typing import List
 
+
+logging.basicConfig(filename='camera_errors.log', level=logging.ERROR,
+                    format='%(asctime)s - Camera %(name)s - %(levelname)s - %(message)s')
+
+class CameraErrorCodes:
+    CAMERA_INITIALIZATION_FAILED = 1001
+    CAPTURE_FAILED = 1002
+    NO_IMAGES_FOUND = 1003
+    READ_FRAME_ERROR = 1004
+    CAMERA_NOT_OPERATIONAL = 1005
+    CONFIGURATION_ERROR = 1006
+
+error_messages = {
+    CameraErrorCodes.CAMERA_INITIALIZATION_FAILED: "Camera initialization failed.",
+    CameraErrorCodes.CAPTURE_FAILED: "Failed to capture image.",
+    CameraErrorCodes.NO_IMAGES_FOUND: "No images found.",
+    CameraErrorCodes.READ_FRAME_ERROR: "Error reading frame.",
+    CameraErrorCodes.CAMERA_NOT_OPERATIONAL: "Camera is not operational.",
+    CameraErrorCodes.CONFIGURATION_ERROR: "Configuration error."
+}
+
+
+class Camera:
+    def __init__(self, camera_id):
+        try:
+            config = self.load_config('../../../configuration/camera_configuration.yml')
+        except Exception as e:
+            logging.error(f"{error_messages[CameraErrorCodes.CONFIGURATION_ERROR]}: {e}")
+            raise ValueError(error_messages[CameraErrorCodes.CONFIGURATION_ERROR])
+            
+        self.camera_id = camera_id
+        os.makedirs(self.image_folder, exist_ok=True)
```

Comment:
no need for file management at the Camera level. This should be done with the CameraManager with the onboard file system